pkg/operation/botanist/controlplane/kube_apiserver_service.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// KubeAPIServiceValues configure the kube-apiserver service.
+type KubeAPIServiceValues struct {
+	Annotations map[string]string
+}
+
+// NewKubeAPIService creates a new instance of DeployWaiter for a specific DNS entry.
+// <waiter> is optional and it's defaulted to github.com/gardener/gardener/pkg/utils/retry.DefaultOps().
+func NewKubeAPIService(
+	values *KubeAPIServiceValues,
+	service client.ObjectKey,
+	sniService *client.ObjectKey,
+	applier kubernetes.ChartApplier,
+	chartsRootPath string,
+	logger *logrus.Entry,
+	client crclient.Client,
+	waiter retry.Ops,
+	clusterIPFunc func(clusterIP string),
+	ingressFunc func(ingressIP string),
+
+) component.DeployWaiter {
+	if waiter == nil {
+		waiter = retry.DefaultOps()
+	}
+
+	if clusterIPFunc == nil {
+		clusterIPFunc = func(_ string) {}
+	}
+
+	if ingressFunc == nil {
+		ingressFunc = func(_ string) {}
+	}
+
+	internalValues := &kubeAPIServiceValues{
+		EnableSNI: sniService != nil && sniService.Name != "" && sniService.Namespace != "",
+		Name:      service.Name,
+	}
+
+	if values != nil {
+		internalValues.Annotations = values.Annotations
+	}
+
+	return &kubeAPIService{
+		ChartApplier:  applier,
+		chartPath:     filepath.Join(chartsRootPath, "seed-controlplane", "charts", "kube-apiserver-service"),
```

Comment:
What do you think about defining a constant and using it in both places here and in line 95?